pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -668,6 +671,7 @@ func templateParams(cfg echo.Config, settings *resource.Settings) (map[string]in
 		"ImageTag":            strings.TrimSuffix(settings.Image.Tag, "-distroless"),
 		"ImagePullPolicy":     settings.Image.PullPolicy,
 		"ImagePullSecretName": imagePullSecretName,
+		"ImageFullPath":       settings.TestEchoImage, // This overrides image hub/tag if it's not empty.
```

Comment:
nit: instead of having multiple variables, just have a single variable `Image` and have it contain the appropriate value based on the settings.  You'll need a `EnableTCPHealthPort` (or something similar) as well.

Reply:
Not sure if I understand this correctly.
Are you suggesting merging `"ImageTag"` and other image related keys in this map to one `"Image"` field?
But this map is also used by `vmDeploymentYaml`, right? Do you want to change that, too?

And I didn't get the comment about `EnableTCPHealthPort`